text/0140-record-feature-flags-on-error.md

```diff
@@ -0,0 +1,45 @@
+- Start Date: 2024-09-04
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/140
+- RFC Status: draft
+
+# Summary
+
+As feature flags are evaluated in a customer's application we will collect those flags and their results, store them in-memory, and push the values to Sentry when an error occurs. The flags can be used to further understand what went wrong during the session.
+
+# Motivation
+
+To enable users to debug application errors with a complete picture of their application's state.
+
+# Options Considered
+
+We will collect flag evaluations and hold them in-memory.  On error event the flags will be placed into the `contexts` object on the `event` and sent to Sentry.
+
+## Transport
+
+The flags will be represented by this data structure during transport. The `flag` key is the name of the flag and the `result` key is the evaluation result returned by the customer's application.
+
+```json
+{
+    "contexts": {
+        "flags": [
```

Comment:
I wonder if we risk conflicting with customers that today already use this key. And if a prefix like `sntr_flags` could be beneficial?


```suggestion
        "sntr_flags": [
```

We don't do this with other contexts though so not sure if it's worth it

Reply:
I'm also wondering if a top-level field will be suggested and what, if any, work that entails. I think its fine to name-space as you did.  I'll leave it as is for now and wait for reviews before changing.

Reply:
@cleptric Should flags exist in the context or should they be given their own top level field?

Reply:
Context

Reply:
What happens if users have already defined this key?